cli/cmd/deploy/deploy.go

```diff
@@ -0,0 +1,397 @@
+package deploy
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+
+	"github.com/AlecAivazis/survey/v2"
+	"github.com/fatih/color"
+	"github.com/go-git/go-git/v5"
+	"github.com/joho/godotenv"
+	"github.com/rilldata/rill/admin/client"
+	"github.com/rilldata/rill/cli/cmd/auth"
+	"github.com/rilldata/rill/cli/cmd/cmdutil"
+	"github.com/rilldata/rill/cli/cmd/org"
+	"github.com/rilldata/rill/cli/cmd/project"
+	"github.com/rilldata/rill/cli/pkg/config"
+	"github.com/rilldata/rill/cli/pkg/dotrill"
+	"github.com/rilldata/rill/cli/pkg/gitutil"
+	"github.com/rilldata/rill/cli/pkg/variable"
+	adminv1 "github.com/rilldata/rill/proto/gen/rill/admin/v1"
+	"github.com/spf13/cobra"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+type options struct {
+	Name        string
+	Description string
+	ProdBranch  string
+	Public      bool
+	Variables   []string
+
+	// projectPath string input taken interactively
+	// no input taken from user for below
+	region   string
+	dbDriver string
+	dbDSN    string
+	slots    int64
+}
```

Comment:
1. Maybe rename to `promptOptions` for clarification?
2. The params that are not prompted for should not be here – instead, use regular CLI flags like in `rill project connect` to take these